scripts/joining_sample_meta_columns.R

```diff
@@ -0,0 +1,60 @@
+library(argparse)
+library(tidyverse)
+source("./src/join_sample_meta.R")
+
+# Argument parser ----
+parser <- ArgumentParser()
+# specify our desired options 
+parser$add_argument('--sample_meta', default= 'sample_meta.csv', help= 'Sample meta data for the sequencing run.')
+parser$add_argument('--cell_set_meta', default= 'cell_set_meta.csv', help= 'Cell set metadata for the sequencing run.')
+parser$add_argument('--l2fc', default= 'l2fc.csv', help= 'L2FC data.') # level 4
+parser$add_argument('--collapsed_l2fc', default= 'collapsed_l2fc.csv', help= 'Collapsed l2fc data.') # level 5
+parser$add_argument('--sig_cols', default= 'cell_set,treatment,dose,dose_unit,day', 
+                    help= 'Columns that uniquely identify a condition.') 
+parser$add_argument('--out', default= getwd(), help= 'Path to the output directory.')
+
+args <- parser$parse_args()
+
+# set output to working directory if none is specified
+if (args$out == "") {
+  args$out = args$wkdir
+}
+
+# Prepare args ----
+sample_meta= data.table::fread(args$sample_meta, header= T, sep= ',')
+cell_set_meta= data.table::fread(args$cell_set_meta, header= T, sep= ',')
+sig_cols= unlist(strsplit(args$sig_cols, ","))
+
+# Add in metadata for l2fc file ----
+if(file.exists(args$l2fc)) {
+  l2fc= data.table::fread(args$l2fc, header= T, sep= ',')
+  if('bio_rep' %in% sample_meta & 'bio_rep' %in% l2fc) {
+    input_cols= c(sig_cols, 'bio_rep')
+  } else {
+    input_cols= sig_cols
+    print('WARNING: No "bio_rep" column detected. Proceeding with just sig_cols.')
+  }
+  l2fc_with_sm= join_sample_meta(df= l2fc, sample_meta, cell_set_meta, key_cols= input_cols)
+  
+  # Write out
+  outpath= paste(args$out, 'l2fc_with_sm.csv', sep='/')
```

Comment:
The default output should be the one with the joined metadata. (the file without the joined sm can be a different name in principle.)
Otherwise we will break the adapter script, since that is looking for l2fc.csv and collapsed_l2fc.csv  files that have these joined columns

Reply:
I would prefer to update the adaptor script. I'll take a look at the adaptor script to see if it is easy to change